lib/rules/prefer-stateless-function.js

```diff
@@ -298,6 +302,20 @@ module.exports = {
         markThisAsUsed(node);
       },
 
+      // Report non-"render" methods that return JSX
+      MethodDefinition: function(node) {
+        if (
+          utils.isES6Component(node.parent.parent) &&
+          node.key.name !== 'render' &&
+          utils.isReturningJSX(node.value)
+        ) {
+          context.report({
+            node: node,
+            message: 'Method returning JSX should be written as stateless function component'
```

Comment:
This should probably be pluralized; `Methods returning JSX should be written as stateless functional components`